packages/amplify-cli/src/amplify-exception-handler.ts

```diff
@@ -37,55 +37,55 @@ export const handleException = async (exception: unknown): Promise<void> => {
     amplifyException = genericErrorToAmplifyException(exception);
   }
 
+  const deepestException = getDeepestAmplifyException(amplifyException);
+  if (context && isHeadlessCommand(context)) {
+    printHeadlessAmplifyException(deepestException);
+  } else {
+    printAmplifyException(deepestException);
+  }
+
   if (context?.usageData) {
     await executeSafely(
-      () => context?.usageData.emitError(amplifyException), 'Failed to emit error to usage data',
+      () => {
+        context?.usageData.emitError(deepestException);
+        printer.blankLine();
+        printer.info(`Session Identifier: ${context?.usageData.getSessionUuid()}`);
+      }, 'Failed to emit error to usage data',
     );
   }
 
-  if (context && isHeadlessCommand(context)) {
-    printHeadlessAmplifyException(amplifyException);
-  } else {
-    printAmplifyException(amplifyException);
-
-    let { downstreamException } = amplifyException;
-    while (isDebug && downstreamException) {
-      printer.blankLine();
-
-      if (downstreamException instanceof AmplifyException) {
-        printAmplifyException(downstreamException);
-        downstreamException = downstreamException.downstreamException;
-      } else {
-        printError(downstreamException);
-        downstreamException = undefined;
-      }
-    }
-  }
-
   // Swallow and continue if any operations fail
   if (context) {
-    await executeSafely(() => reportError(context, amplifyException), 'Failed to report error');
+    await executeSafely(() => reportError(context, deepestException), 'Failed to report error');
   }
 
   await executeSafely(
     () => executeHooks(HooksMeta.getInstance(undefined, 'post', {
-      message: amplifyException.message ?? 'undefined error in Amplify process',
-      stack: amplifyException.stack ?? 'undefined error stack',
+      message: deepestException.message ?? 'undefined error in Amplify process',
+      stack: deepestException.stack ?? 'undefined error stack',
     })),
     'Failed to execute hooks',
   );
 
   await executeSafely(
     () => logger.logError({
-      message: amplifyException.message,
-      error: amplifyException,
+      message: deepestException.message,
+      error: deepestException,
     }),
     'Failed to log error',
   );
 
   process.exitCode = 1;
 };
 
+const getDeepestAmplifyException = (amplifyException: AmplifyException): AmplifyException => {
+  let deepestAmplifyException = amplifyException;
+  while (deepestAmplifyException.downstreamException && amplifyException.downstreamException instanceof AmplifyException) {
```

Comment:
Doesn't the condition need to check `deepestAmplifyException.downstreamException`?

```suggestion
  while (deepestAmplifyException.downstreamException && deepestAmplifyException.downstreamException instanceof AmplifyException) {
```

Reply:
good catch